src/api/informer.rs

```diff
@@ -150,56 +166,33 @@ where
                         let new_version = match &event {
                             Ok(WatchEvent::Added(o))
                             | Ok(WatchEvent::Modified(o))
-                            | Ok(WatchEvent::Deleted(o)) => o.meta().resourceVersion.clone(),
-                            Ok(WatchEvent::Error(e)) => {
-                                // Reset our version to the initial version
-                                // and then return a DesyncError.
-                                // This will terminate the stream early in the
-                                // following `take_while`
-                                *version.write().unwrap() = initial_version.clone();
-                                return Err(crate::Error::KubeDesyncError(e.clone()));
-                            }
+                            | Ok(WatchEvent::Deleted(o)) => o.meta().resourceVersion.clone(), 
                             _ => None,
                         };
 
+                        // If we hit an error, mark that we need to resync on the next call
+                        if let Err(e) = &event {
+                            warn!("Poll error: {:?}", e);
+                            *needs_resync.write().unwrap() = true;
+                        }
```

Comment:
This looks a lot better! Unfortunately, it does not actually deal with resync here!
This is because the 410 Gone error event is an Ok(WatchError:Error(e))!

Reply:
Handle by implementing your suggestions below 👍